wikimedia/wikipedia-ios
# Refresh Notifications Center if push notification is displayed while app is foregrounded

**Phabricator:** https://phabricator.wikimedia.org/T301127

### Notes
This PR adds the ability for the app to refresh notifications and the notifications center if a push alert banner is displayed while the app is foregrounded.

- Adds a notification that posts locally whenever an Echo push notification banner is displayed while the app is foregrounded.
- If a push notification is displayed while the Notifications Center is open, triggers a refresh of the Notifications Center and updates view.
- Adds explicitly refreshing the Notifications Center data when the Settings view appears. Instead of limiting this just to when the user has disabled the Explore feed, I left it on every appearance as it seemed like a reasonable opportunity to attempt to silently refresh the data.
- Notification bell badges should now update to be in the indicated state if receiving a push notification banner while the app is foregrounded. I didn't have to do any specific work for this – it's just a built in benefit of a badge update notification fired down the line from a `loadNotifications` call.

### Test Steps

Mostly lightly generally regression test receiving push notifications while the app is open in different views.

1. Mark all your notifications as read.
2. With the app foregrounded in either the Explore feed or Settings view (with the Explore feed off), trigger a notification from a separate device (like Desktop) and use our Push utility to confirm a) the push banner shows over the foregrounded app and b) the Notifications indicator bell is now in an indicated state. (Note you may have to change the interruption level in  `NotificationService` depending on the notification type to test.)
3. Same as step 3, but navigate to a view that isn't the root Explore feed or Settings view. Confirm the push notification appears, and that when you navigate back to the Explore feed or Settings view that the bell icon is indicated.
